### PR TITLE
Add FlushPreservesDefaultNI and mark RequiresNonDefaultNI

### PR DESCRIPTION
**Pull request by liulk**

### CODE REVIEW

**Comment by xw-g** [comment]
Let's have a quick discuss offline first.

File: compliance/compliance.go [hunk 1]
```diff
@@ -100,6 +100,9 @@ type Test struct {
 	// RequiresImplicitReplace marks a test that requires the implementation of AFTOperation
 	// ADD for entries that already exist.
 	RequiresImplicitReplace bool
+	// RequiresNonDefaultNI marks a test that requires the implementation to configure
+	// entries for non-default network-instance.
+	RequiresNonDefaultNI bool
```
Comment (xw-g) [inline compliance/compliance.go:105]: nit: this seems too generic, how about: "RequiresNHInNonDefaultNI"? same with the flag.

File: cmd/ccli/ccli_test.go [hunk 2]
```diff
@@ -91,19 +108,11 @@ func TestCompliance(t *testing.T) {
...
-			continue
-		}
-
-		if skip := *skipImplicitReplace; skip && tt.In.RequiresImplicitReplace {
-			continue
-		}
-
 		t.Run(tt.In.ShortName, func(t *testing.T) {
+			if reason := shouldSkip(tt); reason != "" {
+				t.Skip(reason)
```
Comment (xw-g) [inline cmd/ccli/ccli_test.go:113]: +1 to improve it with t.Skip!!!

**Review by xw-g — CHANGES_REQUESTED** [review]

File: compliance/compliance.go (hunk 1 above)
Comment (xw-g) [reply compliance/compliance.go:105]: correction: "RequiresNHOnlyInDefault"

Comment (liulk) [reply compliance/compliance.go]: Using the name "RequiresNonDefaultNINHG" to be consistent with "AddIPv4EntryDifferentNINHG"

By the way, although you suggested adding a "flush" version of AddIPv4EntryDifferentNINHG in private chat, I think there is some detail that needs to be fleshed out first. Can we defer that to a follow up PR?

**Review by xw-g — APPROVED** [review]

**Pull request merged**